nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/SetSystemPropertiesAction.java

```diff
@@ -52,14 +65,51 @@ public Topology getTopology(Map<String, String> options) {
 
     public static class SetSystemPropertyComputation extends AbstractBulkComputation {
 
+        public static final String NOTIFY = "param-notify";
+
         public SetSystemPropertyComputation() {
             super(ACTION_NAME);
         }
 
         @Override
         protected void compute(CoreSession session, List<String> ids, Map<String, Serializable> properties) {
-            ids.forEach(id -> properties.forEach((k, v) -> session.setDocumentSystemProp(new IdRef(id), k, v)));
-            session.save();
+            Map<String, Serializable> actualProperties = new HashMap<>(properties);
+            String eventId = (String) actualProperties.remove(NOTIFY);
+            Collection<DocumentRef> refs = ids.stream().map(IdRef::new).collect(Collectors.toList());
+            for (DocumentRef ref : refs) {
+                for (Entry<String, Serializable> entry : actualProperties.entrySet()) {
+                    try {
+                        session.setDocumentSystemProp(ref, entry.getKey(), entry.getValue());
+                    } catch (Exception e) {
+                        // TODO send to error stream
+                        getLog().warn("Cannot set system property: " + entry.getKey() + " on: " + ref.toString(), e);
+                    }
+                }
+            }
+            try {
+                session.save();
+            } catch (PropertyException e) {
+                // TODO send to error stream
+                getLog().warn("Cannot save session", e);
+            }
+            if (eventId != null) {
+                fireEvent(session, eventId, refs);
```

Comment:
I don't think we want to fire event for documents where setDocumentSystemProp fail or if session.save is in error

Reply:
Yeah catch blocs were a quick fix that broke the logic ..
I'm not sure it's good practice to handle exceptions there, maybe it would be better in parent abstract class.